### PR TITLE
Extend MPI support to Cray compilers

### PR DESCRIPTION
**Pull request by victorusu**
This PR changes the MPI search function to support a simpler compiler 
definition list and test the compiler after detection.
Additionally in the PR removes the m4 cache files.

The advantages of this approach is that one can select the list of
compilers to support in a single place, instead of multiple if else
combinations. Furthermore it also tests whether the actual compiler
works by testing it against a simple mpi example program.

There are a bunch of changed files, this is due to the fact that I have 
run `aclocal && autoconf && automake`.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -88,21 +88,19 @@ AM_CONDITIONAL(MPI_ENGINE, test x$mpi_engine = xyes)
...
-    if test x$mpicxx_present = xno; then
-      AC_MSG_ERROR([Fatal Error: no mpi compiler found])
-      exit;
-    else
-      CXX=mpiCC
-    fi
-  else
-    CXX=mpicxx
-  fi
+  AC_CHECK_PROGS(MPICXX, mpicxx mpic++ CC, $MPICXX)
```
Comment (victorusu) [inline configure.ac:92]: This is the main change of this PR.
The rest is just a bunch of files changed because of this change.

File: configure [hunk 2]
```diff
@@ -2808,7 +2808,7 @@ fi
 
 # Define the identity of the package.
  PACKAGE='greasy'
- VERSION='2.2'
+ VERSION='2.2.3'
```
Comment (victorusu) [inline configure:2811]: I took the freedom to bump the version of the project.

**Comment by victorusu** [comment]
@PabloRodenas, @csimarro or @BSC-Support-Team, could any of you review this PR?
What do you guys think?

File: configure (hunk 2 above)
Comment (victorusu) [reply configure:2811]: I can set to any value, just let me know.

**Comment by victorusu** [comment]
If one runs on a Cray this PR using
```
$./configure --enable-slurm 
```
The output is

```
...
checking for srun... yes
checking for bjobs... no
checking for qsub... no
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
...
```

If one runs on a Cray using 
```
$ ./configure --enable-slurm --enable-mpi-engine 
```
Then, the output is 
```
...
checking for srun... yes
checking for bjobs... no
checking for qsub... no
checking for mpicxx... no
checking for mpic++... no
checking for CC... CC
checking whether the MPI command works... yes
checking whether we are using the GNU C++ compiler... yes
checking whether CC accepts -g... yes
checking dependency style of CC... gcc3
...
```

If one runs in a Linux laptop
```
$ ./configure --enable-mpi-engine
```
The output is
```
...
checking for srun... no
checking for bjobs... no
checking for qsub... no
checking for mpicxx... /usr/bin/mpic++
checking whether the MPI command works... yes
checking whether we are using the GNU C++ compiler... yes
checking whether /usr/bin/mpic++ accepts -g... yes
checking dependency style of /usr/bin/mpic++... gcc3
...
```

File: configure (hunk 2 above)
Comment (BSC-Support-Team) [reply configure:2811]: it is fine. Thanks for sharing your effort!

File: configure.ac (hunk 1 above)
Comment (BSC-Support-Team) [reply configure.ac:92]: I see

**Comment by BSC-Support-Team** [comment]
Dear Victor,

I reviewed the changes and I do not see any problem for accepting the PR. Thanks for the interest on the project. On the other side, we detected a memory leak error on regex file. I just upload our changes to github today, please, review them in your fork.

Best,
Pablo.

**Comment by victorusu** [comment]
> Dear Victor,
> 
> I reviewed the changes and I do not see any problem for accepting the PR. Thanks for the interest on the project. On the other side, we detected a memory leak error on regex file. I just upload our changes to github today, please, review them in your fork.
> 
> Best,
> Pablo.

Dear Pablo,

Thanks for informing us about this!

Best wishes,
Victor

**Pull request merged**